tests/composite_pk/test_checks.py

```diff
@@ -0,0 +1,113 @@
+from django.core import checks
+from django.db import models
+from django.test import TestCase
+from django.test.utils import isolate_apps
+
+from .models import Tenant
+
+
+@isolate_apps("composite_pk")
+class CompositePKChecksTests(TestCase):
+    """
+    Test the system checks related to Meta.primary_key.
+    """
+
+    def test_must_be_a_tuple_or_list_of_strings(self):
+        test_cases = [
+            [],
+            (),
+            (1,),
+            ("id", False),
+        ]
+
+        for i, fields in enumerate(test_cases):
+            with self.subTest(fields=fields):
+                model = type(
+                    f"Model{i}",
+                    (models.Model,),
+                    {
+                        "__module__": Tenant.__module__,
+                        "Meta": type("Meta", (), {"primary_key": fields}),
+                    },
+                )
+
+                self.assertEqual(
+                    model.check(databases=self.databases),
+                    [
+                        checks.Error(
+                            "The value of 'Meta.primary_key' must be a non-empty tuple "
+                            "or list of strings.",
+                            obj=model,
+                            id="models.E042",
+                        ),
+                    ],
+                )
+
+    def test_field_may_not_set_primary_key(self):
+        class Model(models.Model):
+            account_id = models.IntegerField()
+            id = models.IntegerField(primary_key=True)
+
+            class Meta:
+                primary_key = ("account_id", "id")
+
+        self.assertEqual(
+            Model.check(databases=self.databases),
+            [
+                checks.Error(
+                    "id may not set primary_key=True if Meta.primary_key is defined.",
+                    obj=Model,
+                    id="models.E043",
+                ),
+            ],
+        )
+
+    def test_may_only_contain_attnames(self):
+        class Parent(models.Model):
+            pass
+
+        class Child(models.Model):
+            parent = models.ForeignKey(Parent, on_delete=models.CASCADE)
+            id = models.SmallIntegerField()
+
+            class Meta:
+                primary_key = ("parent", "id")
+
+        self.assertEqual(
+            Child.check(databases=self.databases),
+            [
+                checks.Error(
+                    "Meta.primary_key may only reference concrete fields.",
+                    obj=Child,
+                    id="models.E044",
+                ),
+            ],
+        )
```

Comment:
How much work would it be to allow this? This feels like a pretty natural thing for a user to want to do.

I don't think this is a blocker though - I think relaxing the API here is something that could be done as a follow-up PR.

Reply:
We could allow it for foreign keys, so:

`primary_key = ("parent", "id")` means `primary_key = ("parent_id", "id")`,

But IMO it's clearer this way (and also simpler to implement).

If we're going to implement composite foreign keys later, then it would look like this:

`primary_key = ("parent", "id")` means `primary_key = ("account_id", "parent_id", "id")`,

And I'm not sure if this is a good direction.

Reply:
Done 👍